File: backend/app/core/sentry.py
"""Optional Sentry init for production (no hard crash if package missing)."""

from __future__ import annotations

from loguru import logger


def init_sentry(*, dsn: str, environment: str) -> None:
    if not dsn:
        return
    try:
        import sentry_sdk
        from sentry_sdk.integrations.fastapi import FastApiIntegration
        from sentry_sdk.integrations.starlette import StarletteIntegration

        sentry_sdk.init(
            dsn=dsn,
            environment=environment,
            traces_sample_rate=0.1 if environment == "production" else 0.0,
            integrations=[
                StarletteIntegration(transaction_style="endpoint"),
                FastApiIntegration(transaction_style="endpoint"),
            ],
            send_default_pii=False,
        )
        logger.info("sentry_initialized env={}", environment)
    except Exception as exc:  # noqa: BLE001 — optional dependency
        logger.warning("sentry_init_skipped err={}", str(exc))

File: frontend/src/lib/sentry.ts
/**
 * Optional Sentry bootstrap (TZ §12).
 * Loads @sentry/react only when VITE_SENTRY_DSN is set — no hard dependency crash in local dev.
 */

export async function initSentry(): Promise<void> {
  const dsn = import.meta.env.VITE_SENTRY_DSN;
  if (!dsn) {
    return;
  }
  try {
    const Sentry = await import("@sentry/react");
    Sentry.init({
      dsn,
      environment: import.meta.env.MODE,
      tracesSampleRate: 0.1,
      integrations: [Sentry.browserTracingIntegration()],
    });
  } catch (err) {
    // Package may be absent until npm i @sentry/react
    if (import.meta.env.DEV) {
      console.warn("[sentry] init skipped:", err);
    }
  }
}
